bindings/java/src/org/sleuthkit/datamodel/Blackboard.java

```diff
@@ -237,6 +232,65 @@ public AnalysisResult newAnalysisResult(BlackboardArtifact.Type artifactType, lo
 		}
 	}
 	
+	/**
+	 * Get all analysis results matching the given where sub-clause.
+	 * 
+	 *
+	 * @param whereClause Where sub clause, specifies conditions to match.  
+	 * @return list of analysis results.
+	 *
+	 * @throws TskCoreException exception thrown if a critical error occurs
+	 *                          within TSK core
+	 */
+	public List<AnalysisResult> getAnalysisResultsWhere(String whereClause) throws TskCoreException {
+
+		final String queryString = ANALYSIS_RESULT_QUERY_STRING
+				+ " AND " + whereClause;
+							
+		caseDb.acquireSingleUserCaseReadLock();
+		try (CaseDbConnection connection = caseDb.getConnection();
+				Statement statement = connection.createStatement();
+				ResultSet resultSet = connection.executeQuery(statement, queryString);) {
+			
+			List<AnalysisResult> analysisResults = resultSetToAnalysisResults(resultSet);
+			return analysisResults;
+		} catch (SQLException ex) {
+			throw new TskCoreException(String.format("Error getting analysis results for WHERE caluse = '%s'", whereClause), ex);
+		} finally {
+			caseDb.releaseSingleUserCaseReadLock();
+		}
+	}
+	
+	/**
+	 * Creates AnalysisResult objects for the result set of a
+	 *  table query of the form "SELECT * FROM
+	 * blackboard_artifacts JOIN WHERE XYZ".
+	 *
+	 * @param rs A result set from a query of the blackboard_artifacts table of
+	 *           the form "SELECT * FROM blackboard_artifacts WHERE XYZ".
+	 *
+	 * @return A list of BlackboardArtifact objects.
+	 *
+	 * @throws SQLException     Thrown if there is a problem iterating through
+	 *                          the result set.
+	 * @throws TskCoreException Thrown if there is an error looking up the
+	 *                          artifact type id
+	 */
+	private List<AnalysisResult> resultSetToAnalysisResults(ResultSet resultSet) throws SQLException, TskCoreException {
```

Comment:
Please give the method docs a consistent style. These become part of online API documentation and a consistent style is a good thing in that setting. Personally, I prefer using sentence capitalization and punctuation. This is what was done for the method docs added to Content.